### ISSUE TITLE
read.argo() might mix up columns

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#877]
I realized in working on #833 (see comments there) that we cannot rely on the columns of argo data corresponding to the same variables for each row. This is quite surprising and I don't know whether it is something to worry about, because I'm seeing that in a file that also has a column name that is blank. I plan to get to the bottom of this in #833 and return here only once I've solved both issues.

Mainly I'm making this an issue so CR can see it. He was the one who gave me the link to the file that seems to be odd. I wonder, @richardsc, is this file known to be provisional, or thought to be suitable for use?

**Comment by dankelley** [comment]
Here are my notes, which I was putting into #833 but I realize belong here instead.
Hm. I realized two things about `5904124_20160104211204622.nc`, a file containing oxygen. A clue is now provided by [833e.R](https://github.com/dankelley/oce-issues/blob/master/833/833e.R), which shows
```r
5904124_20160104211204622.nc 
    column sampling (unequal for 5904124_20160104211204622.nc)
    row 1  ' PSAL', 'DOXY', 'PRES', 'TEMP', ' '
    row 4  ' PSAL', 'DOXY', 'TEMP', 'PRES', ' '
```
The odd things are as follows.

1. the columns are not in the same order. This makes me worry about previous code, so I've made a new issue (#873) for that.

2. the fifth column has no name. From the output below, I think it must be `TEMP_DOXY`. Maybe that's not an official name, so it is given as blank?
```
ncdump -h ~/Dropbox/oce-data/argo/5904124_20160104211204622.nc|grep float|grep '.*(N_PROF'
	float PRES(N_PROF, N_LEVELS) ;
	float PRES_ADJUSTED(N_PROF, N_LEVELS) ;
	float PRES_ADJUSTED_ERROR(N_PROF, N_LEVELS) ;
	float PSAL(N_PROF, N_LEVELS) ;
	float PSAL_ADJUSTED(N_PROF, N_LEVELS) ;
	float PSAL_ADJUSTED_ERROR(N_PROF, N_LEVELS) ;
	float DOXY(N_PROF, N_LEVELS) ;
	float DOXY_ADJUSTED(N_PROF, N_LEVELS) ;
	float DOXY_ADJUSTED_ERROR(N_PROF, N_LEVELS) ;
	float TEMP(N_PROF, N_LEVELS) ;
	float TEMP_ADJUSTED(N_PROF, N_LEVELS) ;
	float TEMP_ADJUSTED_ERROR(N_PROF, N_LEVELS) ;
	float TEMP_DOXY(N_PROF, N_LEVELS) ;
	float TEMP_DOXY_ADJUSTED(N_PROF, N_LEVELS) ;
	float TEMP_DOXY_ADJUSTED_ERROR(N_PROF, N_LEVELS) ;
```

**Comment by dankelley** [comment]
The blank names are because things are not always filled in. So that mystery is solved.

Another mystery: `TEMP_DOXY` occurs in **JUST ONE** spot, profile 99. I had been thinking this file was profiles of one float. Maybe it's several floats?
```r
> t(N[,97:110])
      [,1]   [,2]   [,3]   [,4]        [,5]  
 [1,] "TEMP" "PRES" "PSAL" ""          ""    
 [2,] "PRES" "TEMP" "PSAL" ""          ""    
 [3,] "PRES" "TEMP" "PSAL" "TEMP_DOXY" "DOXY"
 [4,] "TEMP" "PRES" "PSAL" ""          ""    
 [5,] "TEMP" "PRES" "PSAL" ""          ""    
 [6,] "PRES" "TEMP" "PSAL" ""          ""    
 [7,] "PRES" "TEMP" "PSAL" ""          ""    
 [8,] "PSAL" "TEMP" "PRES" ""          ""    
 [9,] "PSAL" "DOXY" "TEMP" "PRES"      ""    
[10,] "PSAL" "DOXY" "TEMP" "PRES"      ""    
[11,] "PSAL" "DOXY" "TEMP" "PRES"      ""    
[12,] "PRES" "TEMP" "PSAL" "DOXY"      ""    
[13,] "TEMP" "PRES" "PSAL" ""          ""    
[14,] "PRES" "PSAL" "TEMP" ""          ""    
```

**Comment by dankelley** [comment]
I don't think this happens, and also I'm updating `read.argo` significantly, so I'm closing this.
